pkg/sourcereader/embedded.go

```diff
@@ -15,95 +15,35 @@
 package sourcereader
 
 import (
+	"embed"
 	"fmt"
 	"hpc-toolkit/pkg/modulereader"
-	"io/fs"
-	"io/ioutil"
-	"os"
-	"path"
-	"path/filepath"
+
+	"github.com/hashicorp/terraform-config-inspect/tfconfig"
 )
 
 // ModuleFS contains embedded modules (./modules) for use in building
 // blueprints. The main package creates and injects the modules directory as
 // hpc-toolkit/modules are not accessible at the package level.
-var ModuleFS BaseFS
-
-// BaseFS is an extension of the io.fs interface with the functionality needed
-// in CopyDirFromModules. Works with embed.FS and afero.FS
-type BaseFS interface {
-	ReadDir(string) ([]fs.DirEntry, error)
-	ReadFile(string) ([]byte, error)
-}
+var ModuleFS embed.FS
 
 // EmbeddedSourceReader reads modules from a local directory
 type EmbeddedSourceReader struct{}
 
-// copyDirFromModules copies an FS directory to a local path
-func copyDirFromModules(fs BaseFS, source string, dest string) error {
-	dirEntries, err := fs.ReadDir(source)
-	if err != nil {
-		return err
-	}
-	for _, dirEntry := range dirEntries {
-		entryName := dirEntry.Name()
-		// path package (not path/filepath) should be used for embedded source
-		// as the path separator is a forward slash, even on Windows systems.
-		// https://pkg.go.dev/embed#hdr-Directives
-		entrySource := path.Join(source, entryName)
-		entryDest := filepath.Join(dest, entryName)
-		if dirEntry.IsDir() {
-			if err := os.Mkdir(entryDest, 0755); err != nil {
-				return err
-			}
-			if err = copyDirFromModules(fs, entrySource, entryDest); err != nil {
-				return err
-			}
-		} else {
-			fileBytes, err := fs.ReadFile(entrySource)
-			if err != nil {
-				return err
-			}
-			copyFile, err := os.Create(entryDest)
-			if err != nil {
-				return err
-			}
-			if _, err = copyFile.Write(fileBytes); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-// copyFSToTempDir is a temporary workaround until tfconfig.ReadFromFilesystem
-// works against embed.FS.
-// Open Issue: https://github.com/hashicorp/terraform-config-inspect/issues/68
-func copyFSToTempDir(fs BaseFS, modulePath string) (string, error) {
-	tmpDir, err := ioutil.TempDir("", "tfconfig-module-*")
-	if err != nil {
-		return tmpDir, err
-	}
-	err = copyDirFromModules(fs, modulePath, tmpDir)
-	return tmpDir, err
-}
-
 // GetModuleInfo gets modulereader.ModuleInfo for the given kind from the embedded source
 func (r EmbeddedSourceReader) GetModuleInfo(modPath string, kind string) (modulereader.ModuleInfo, error) {
 	if !IsEmbeddedPath(modPath) {
 		return modulereader.ModuleInfo{}, fmt.Errorf("Source is not valid: %s", modPath)
 	}
 
-	modDir, err := copyFSToTempDir(ModuleFS, modPath)
-	defer os.RemoveAll(modDir)
-	if err != nil {
-		err = fmt.Errorf("failed to copy embedded module at %s to tmp dir %s: %v",
-			modPath, modDir, err)
+	mod, diag := tfconfig.LoadModuleFromFilesystem(tfconfig.WrapFS(ModuleFS), modPath)
```

Comment:
This needs to be reorganized a bit. `tfconfig` shouldn't be used directly in the `sourcereader` package, but instead in the `modulereader` package.

`sourcereader` essentially handles a lot of the IO and access to the raw contents, `modulereader` is specific to a module kind (terraform, packer, etc) and is able to provide more detailed information like Inputs/Outputs/etc of the module being referenced.

This function should pass the `modPath` into [`reader.GetInfo()`](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/pkg/modulereader/tfreader.go#L32), which calls [`hcl_utils.getHCLInfo()`](https://github.com/GoogleCloudPlatform/hpc-toolkit/blob/main/pkg/modulereader/hcl_utils.go#L27). The `getHCLInfo()` function can now check if it's embedded or not. If it isn't, it will do what it does today. If it is, it will call `LoadModuleFromFilesystem` rather than `LoadModule`.

The flow will look like this:
GetModuleInfo() -> reader.GetInfo() -> getHCLInfo() -> LoadModule() for local or LoadModuleFromFilesystem() for embedded